lib/src/cpp/domain_participant/DomainParticipant.cpp

```diff
@@ -20,314 +20,437 @@
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void clear_participant_id(
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void clear_user_transports(
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void clear_user_transports(
         const std::string& xml_file,
         const std::string& profile_id,
         int32_t index)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void clear_use_builtin_transports(
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void clear_user_data(
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void clear_user_data(
         const std::string& xml_file,
         const std::string& profile_id,
         int32_t index)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void clear_prefix(
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void set_default_profile(
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void set_domain_id(
         const std::string& xml_file,
         const std::string& profile_id,
         const std::string& domain_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void set_name(
         const std::string& xml_file,
         const std::string& profile_id,
         const std::string& name)
 {
-    throw Unsupported("Unsupported");
+    // Xerces objects to manage XML elements
+    xercesc::DOMConfiguration* config = NULL;
+    xercesc::DOMDocument* doc = NULL;
+    ParseXMLErrorHandler* error_handler = NULL;
+    xercesc::DOMImplementation* implementation = NULL;
+    xercesc::DOMLSOutput* output = NULL;
+    xercesc::XercesDOMParser* parser = NULL;
+    xercesc::DOMLSSerializer* serializer = NULL;
+    xercesc::XMLFormatTarget* target = NULL;
+
+    // XML nodes and values
+    xercesc::DOMNode* profiles_node = NULL;
+    xercesc::DOMNode* participant_node = NULL;
+    xercesc::DOMNode* rtps_node = NULL;
+    xercesc::DOMNode* name_node = NULL;
+    xercesc::DOMText* name_value = NULL;
```

Comment:
I suppose we should have a class with all these variables that can be accessed by every function. Otherwise defining all these variables in each function is going to take a lot of code lines.